airflow/providers/amazon/aws/operators/redshift_cluster.py

```diff
@@ -624,8 +629,15 @@ def __init__(
         self._attempts = 10
         self._attempt_interval = 15
         self.redshift_hook = RedshiftHook(aws_conn_id=aws_conn_id)
+        self.aws_conn_id = aws_conn_id
+        self.deferrable = deferrable
+        self.max_attempts = max_attempts
+
+        if self.deferrable and self.wait_for_completion:
+            raise AirflowException("Both wait_for_completion and deferrable can't be True.")
```

Comment:
I was thinking more on this, and I'm still not sure what the best practice would be. On one hand, I understand what you are saying here - both `deferrable` and `wait_for_completion` shouldn't be `True` because only one path can be taken (in this case, the `deferrable` path will be taken). 

But on the other hand, is it important to bother the user with that detail? If a user wants an operator to complete its execution, and also wants deferrable behavior, we can make the choice for the user use the deferrable parameter.

Reply:
We followed this review suggestion from @kaxil last time in dbt operators w.r.t `wait_for_completion` vs `deferrable`. I suggest to follow the same

https://github.com/apache/airflow/pull/29014#discussion_r1073723719

Reply:
That's a good suggestion. I agree

Reply:
removed it